docs/source/developers/python.rst

```diff
@@ -636,6 +636,52 @@ Caveats
 
 The Plasma component is not supported on Windows.
 
+Deleting stale build artifacts
+==============================
+
+When there have been changes to the structure of the Arrow C++ library or Pyarrow,
+a thorough cleaning is recommended as a first attempt to fixing build errors.
+
+.. note::
+
+   It is not necessarily intuitive from the error itself that the problem is due to stale artifacts.
+   Example of a build error from stale artifacts is "Unknown CMake command "arrow_keep_backward_compatibility".
```

Comment:
```suggestion
   Example of a build error from stale artifacts is "Unknown CMake command "arrow_keep_backward_compatibility"".
```

Or we can use a ReST markup instead of `"..."`.